core/src/main/scala/cats/instances/partialFunction.scala

```diff
@@ -0,0 +1,60 @@
+package cats.instances
+import cats.arrow.{ArrowChoice, CommutativeArrow}
+
+trait PartialFunctionInstances {
+
+  implicit val catsStdInstancesForArrowHierarchy: ArrowChoice[PartialFunction] with CommutativeArrow[PartialFunction] =
+    new ArrowChoice[PartialFunction] with CommutativeArrow[PartialFunction] {
+
+      /**
+       * {{{
+       * scala> import cats.arrow.Arrow
+       * scala> import cats.syntax.arrowChoice._
+       * scala> val toLong: PartialFunction[Int, Long] = Arrow[PartialFunction].lift(_.toLong)
+       * scala> val toDouble: PartialFunction[Float, Double] = Arrow[PartialFunction].lift(_.toDouble)
+       * scala> val f: PartialFunction[Either[Int, Float], Either[Long, Double]] = toLong +++ toDouble
+       * scala> f(Left(3))
+       * res0: Either[Long,Double] = Left(3)
+       * scala> f(Right(3))
+       * res1: Either[Long,Double] = Right(3.0)
+       * }}}
+       */
+      override def choose[A, B, C, D](f: PartialFunction[A, C])(
+        g: PartialFunction[B, D]
+      ): PartialFunction[Either[A, B], Either[C, D]] = {
+        case Left(a) if f.isDefinedAt(a)  => Left(f(a))
+        case Right(b) if g.isDefinedAt(b) => Right(g(b))
+      }
+
+      override def lift[A, B](f: A => B): PartialFunction[A, B] = { case a if a.isInstanceOf[A] => f(a) }
```

Comment:
Could this be ?
```suggestion
      override def lift[A, B](f: A => B): PartialFunction[A, B] = { case a => f(a) }
```

Reply:
yes, this works. 

I originally tried `{ case a: A => f(a) }` which led to this type erasure error-

> [error] /Users/gagandeepkalra/IdeaProjects/cats/core/src/main/scala/cats/instances/partialFunction.scala:29:77: abstract type pattern A is unchecked since it is eliminated by erasure
[error]       override def lift[A, B](f: A => B): PartialFunction[A, B] = { case a: A => f(a) }
[error]    

As I understand adding `: A` introduces another type check (which is not possible because of erasure), but the function `{ case a => f(a) }` in itself type checks at compile team no runtime check needed?